src/DIRAC/DataManagementSystem/Client/DataManager.py

```diff
@@ -160,24 +160,22 @@ def __cleanDirectory(self, folder):
             errStr = "Write access not permitted for this credential."
             log.debug(errStr, folder)
             return S_ERROR(errStr)
-        res = self.__getCatalogDirectoryContents([folder], includeDirectories=True)
+
+        res = returnSingleResult(self.fileCatalog.getDirectoryDump([folder]))
+
         if not res["OK"]:
             return res
 
-        if not res["Value"]:
+        if not (res["Value"]["Files"] or res["Value"]["SubDirs"]):
             # folder is empty, just remove it and return
             res = returnSingleResult(self.fileCatalog.removeDirectory(folder, recursive=True))
```

Comment:
I'm surprised that this has `recursive=True`?

Reply:
It's maybe not needed indeed, but I'd rather not change it here in the risk of breaking something, and I'd like this to be merged before Monday's release :angel: